### DISCUSSION TITLE
hocuspocus is advertised as open source, where is the code ?

### DISCUSSION THREAD

**Discussion by gaetan-wap** [BitPhinix/slate-yjs#368] [General]
Hello,

I am looking for hocuspocus source  code. It is referenced as open source on hocuspocus.dev?

Links referenced on packages npm documentation are dead: https://github.com/ueberdosis/hocuspocus.
I could not find anything more

Any idea where the code could be ?

Thanks

**Comment by BitPhinix** [comment]
It's currently sponsorware, you have to donate to get access, but it will be fully oss once it's reached the 1.0 release. Seeing as there is currently a 1.0.0-beta release, it should hopefully be very soon 🙏.

Keep in mind that you don't have to use hocuspocus though, slate-yjs itself is provider agnostic so you can also use something like [y-websocket](https://github.com/yjs/y-websocket) to get going.

**Reply by gaetan-wap** [reply]
Thanks for your answer.

I am already using `y-websocket` I wanted to give a try to hoctupocus to see if it reduces issues on mobile phone (letters, words get duplicated while typing) when used in conjunction with Plate.

**Reply by BitPhinix** [reply]
I don't think the persistence layer is the issue there 😓 It's very likely the soft keyboard on android causing the issue as it can't deal with node changes (caused by e.g. remote changes) while composing. At least, that is what I would suspect the cause to be here.

**Reply by gaetan-wap** [reply]
When I try with slate-js example, behaviour is better. That why I suspect Plate, persistence layer, or my integration.
